src/Router.ts

```diff
@@ -318,11 +319,10 @@ export class Router<C extends Context> extends Evented {
 						//
 						// Reset selected routes if not dispatched from start().
 						this._currentSelection = dispatchFromStart ? result : [];
-
-						for (const { handler, params } of result) {
-							catchRejection(this, context, path, handler({ context, params }));
-						}
-
+						const promise = result.reduce((promise, { handler, params }) => {
```

Comment:
Is there really some sort of implicit order in the handlers?  Would this not work here?

```ts
const promise = Promise.all(result.map(({ handler, params }) => handler({ context, params })));
```